snapcraft/internal/errors.py

```diff
@@ -168,39 +182,44 @@ def __init__(self, filename, message):
 class SnapcraftPartMissingError(SnapcraftError):
 
     fmt = (
-        'Cannot find the definition for part {part_name!r}.\n'
-        'It may be a remote part, run `snapcraft update` '
-        'to refresh the remote parts cache.'
+        'Failed to get part information: '
+        'Cannot find the definition for part {part_name!r}. '
+        'If it is a remote part, run `snapcraft update` '
+        'to refresh the remote parts cache. '
+        'If it is a local part, make sure that it is defined in the '
+        '`snapcraft.yaml`.'
```

Comment:
Which case here do we feel is more likely: a remote part being used with an out-of-date cache, or the user misspelling a part (or forgetting to put it in, etc.)? Whichever one is more likely, we should probably say first. I kinda feel like the latter is more likely, so perhaps the order here should be switched?

Reply:
I would recommend the most important go last actually. As a general rule of thumb that's what stands out more when you read a text.